http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -282,6 +262,59 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
       )
     } yield ()
 
+  def startRuntime(userInfo: UserInfo, runtimeName: RuntimeName, workspaceId: WorkspaceId)(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- as.ask
+    runtime <- RuntimeServiceDbQueries.getActiveRuntimeRecord(workspaceId, runtimeName).transaction
+
+    // If user is creator of the runtime, they should definitely be able to see the runtime.
+    hasPermission <- checkPermission(runtime.auditInfo.creator,
+                                     userInfo,
+                                     runtime.cloudContext,
+                                     runtime.runtimeName,
+                                     runtime.id
+    )
+    _ <- F
+      .raiseError[Unit](
+        RuntimeNotFoundException(runtime.cloudContext, runtimeName, "permission denied", Some(ctx.traceId))
+      )
+      .whenA(!hasPermission)
+    _ <-
+      if (runtime.status.isStartable) F.unit
+      else
+        F.raiseError[Unit](RuntimeCannotBeStartedException(runtime.cloudContext, runtime.runtimeName, runtime.status))
+    _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.PreStarting, ctx.now).transaction
+    _ <- publisherQueue.offer(StartRuntimeMessage(runtime.id, Some(ctx.traceId)))
```

Comment:
If I'm reading this correctly, 287 and 288 should be on line 284

Reply:
why do you think that way?

Reply:
oh I guess you are thinking of sth like
```
if (condition)
 do X
else
 do y
```

But it's quite common in Scala, we do
```
for {
 _ <- someCheck, and if this fails, terminate the flow
 _ <- happy path flow
} yield ()
```

Reply:
I misread the indentation!

It looks good